### PR TITLE
Add HTTP username and password, ca-cert and no-verify parameters

### PR DESCRIPTION
**Pull request by sadok-f**
In order to benchmark an elasticsearch cluster which is running in SSL and with X-pack security enabled (HTTP authentification) I make this PR to:
- Add HTTP username and password parameters.
- Add ca-cert parameter
- Add no-verify parameter

### CODE REVIEW

File: elasticsearch-stress-test.py [hunk 1]
```diff
@@ -54,9 +59,15 @@
 parser.add_argument("--no-cleanup", default=False, action='store_true', help="Don't delete the indices upon finish")
 parser.add_argument("--stats-frequency", type=int, default=30,
                     help="Number of seconds to wait between stats prints (default 30)")
-parser.add_argument("--not-green", dest="green", action="store_false")
+parser.add_argument("--not-green", dest="green", action="store_false", help="Script doesn't wait for the cluster to be green")
 parser.set_defaults(green=True)
 
+parser.add_argument("--ca-file", dest="cafile", default="", help="Path to your certificate file")
+parser.add_argument("--no-verify", default=True, action="store_false", help="Do not verify certificate")
```
Comment (roiravhon) [inline elasticsearch-stress-test.py]: Should default be false?

File: elasticsearch-stress-test.py [hunk 2]
```diff
@@ -335,10 +356,23 @@ def main():
...
+
+            if CA_FILE:
+                context = create_ssl_context(cafile=CA_FILE)
+         
+            if AUTH_USERNAME and AUTH_PASSWORD:
+                auth = (AUTH_USERNAME, AUTH_PASSWORD)
+           
+            es = Elasticsearch(
+                esaddress,
+                http_auth=auth,
```
Comment (roiravhon) [inline elasticsearch-stress-test.py]: Can this be None?

File: elasticsearch-stress-test.py [hunk 3]
```diff
@@ -335,10 +356,23 @@ def main():
...
+                verify_certs=VERIFY_CERTS,
+                ssl_context=context,
```
Comment (roiravhon) [inline elasticsearch-stress-test.py:370]: And this?

**Review by roiravhon — COMMENTED** [review]
Thanks for this!
Some comments here

File: elasticsearch-stress-test.py (hunk 1 above)
Comment (sadok-f) [reply elasticsearch-stress-test.py]: Yes you're right it should be False.

File: elasticsearch-stress-test.py (hunk 2 above)
Comment (sadok-f) [reply elasticsearch-stress-test.py]: it's declared in the line 345: auth = None

File: elasticsearch-stress-test.py (hunk 3 above)
Comment (sadok-f) [reply elasticsearch-stress-test.py:370]: And this one also line 346: context = None

**Comment by roiravhon** [comment]
Great. I assume you tested both SSL and none-SSL scenarios with the new code, right?

**Comment by sadok-f** [comment]
Yea both scenarios work well.

**Comment by roiravhon** [comment]
Awesome. Thanks for that!

**Comment by sadok-f** [comment]
Great !

**Pull request merged**